### ISSUE TITLE
Version check in pre gen hook is not being run

### ISSUE THREAD

**Issue by hackebrot** [pydanny/cookiecutter-django#534]
Hi,

this is a follow-up to our gitter conversation about the version check that @pydanny added recently.

Unfortunately **pre_gen_project** hooks in **Cookiecutter** are being run after the context is fully populated. This template uses ``now`` in one of the context variables and hence breaks for Cookiecutter older than 1.4.0.

```bash
ccdj [raphael@hackebook ccdj]$ cookiecutter -V
Cookiecutter 1.3.0 from /Users/raphael/.virtualenvs/ccdj/lib/python3.5/site-packages (Python 3.5)
```

```text
ccdj [raphael@hackebook ccdj]$ cookiecutter gh:pydanny/cookiecutter-django
Cloning into 'cookiecutter-django'...
remote: Counting objects: 5892, done.
remote: Compressing objects: 100% (84/84), done.
remote: Total 5892 (delta 43), reused 7 (delta 7), pack-reused 5801
Receiving objects: 100% (5892/5892), 2.52 MiB | 1.49 MiB/s, done.
Resolving deltas: 100% (3698/3698), done.
Checking connectivity... done.
project_name [project_name]: poyo
project_slug [poyo]:
author_name [Your Name]: Raphael
email [raphael@hackebrot.de]:
description [A short description of the project.]: Nope
domain_name [example.com]:
version [0.1.0]:
timezone [UTC]:
Traceback (most recent call last):
  File "/Users/raphael/.virtualenvs/ccdj/bin/cookiecutter", line 11, in <module>
    sys.exit(main())
...
jinja2.exceptions.TemplateSyntaxError: Encountered unknown tag 'now'.
```

**Comment by pydanny** [comment]
After reflecting on it, and trying it myself, you are correct. Yet the number of requests for help is dropping. Here are some thoughts/ideas:

1. In Cookiecutter core, add a `_cookiecutter_version` setting in `cookiecutter.json` that specifies the minimum Cookiecutter version
2. Don't worry about it. The number of help requests for this is rapidly dropping. Maybe this isn't a big deal afterall.

**Comment by hackebrot** [comment]
I was hoping to have a ``requirements`` field in the new ``context`` format, if any of the declared dependencies is missing, exit gracefully without generating the project.

No, it's definitely not a big problem (I never get any emails. Doubt anyone knows I'm working on Cookiecutter 😉 )

**Comment by jayfk** [comment]
> I never get any emails.

You don't have them enabled in your GitHub profile (wise choice btw).

**Comment by hackebrot** [comment]
@jayfk, I do get github notifications for https://github.com/audreyr/cookiecutter. But I'm not on the collaborator team for **cookiecutter-django**, where it seems to be more common for people to ask questions and report issues.

I rarely get any emails to my personal email address though.

**Comment by jayfk** [comment]
Yeah, I know. 

I think @pydanny was referring to personal emails he gets to the address listed on his GitHub profile (not notifications). That's why I mentioned it.

> But I'm not on the collaborator team for cookiecutter-django, where it seems to be more common for people to ask questions and report issues.

I'll make sure to keep an eye on that and point people in the right direction when I see something like that.

**Comment by hackebrot** [comment]
Right. I'd personally not respond to emails when there's a GitHub issue tracker and a Gitter room.

We are volunteers and there's no point in solving problems multiple times, so anything that helps users from the community should be discussed in the public, IMHO.

**Comment by pydanny** [comment]
Let's nuke that code.
